experiments/SlabLand/balance_law_interface_atmos.jl

```diff
@@ -0,0 +1,197 @@
+import ClimateMachine.BalanceLaws:
+    # declaration
+    vars_state,
+    # initialization
+    init_state_prognostic!,
+    init_state_auxiliary!,
+    nodal_init_state_auxiliary!,
+    # rhs computation
+    compute_gradient_argument!,
+    compute_gradient_flux!,
+    flux_first_order!,
+    flux_second_order!,
+    source!,
+    # boundary conditions
+    boundary_conditions,
+    boundary_state!
+
+import ClimateMachine.DGMethods.NumericalFluxes:
+    CentralNumericalFluxGradient,
+    CentralNumericalFluxSecondOrder,
+    NumericalFluxFirstOrder,
+    NumericalFluxSecondOrder,
+    RusanovNumericalFlux,
+    numerical_boundary_flux_second_order!, 
+    numerical_flux_second_order!, 
+    numerical_boundary_flux_first_order!, 
+    numerical_flux_first_order!,
+    normal_boundary_flux_second_order!
+
+##
+#     Declaration of variables
+##
+"""
+    vars_state 
+    - returns a NamedTuple of data types.
+"""
+function vars_state(model::ModelSetup, aux::Auxiliary, T)
+
+    @vars begin
+        x::T
+        y::T
+        z::T
+        T_sfc::T  # land surface temperature
+    end
+end
+
+function vars_state(::ModelSetup, ::Prognostic, T)
+    @vars begin
+        ρ::T
+        ρu::SVector{3,T}
+        ρθ::T
+        F_ρθ_accum::T
+    end
+end
+
+function vars_state(::ModelSetup, ::Gradient, T)
+    @vars begin
+        ∇ρ::T
+        ∇u::SVector{3,T}
+        ∇θ::T
+    end
+end
+
+function vars_state(::ModelSetup, ::GradientFlux, T)
+    @vars begin
+        μ∇ρ::SVector{3,T}
+        ν∇u::SMatrix{3,3,T,9}
+        κ∇θ::SVector{3,T}
+    end
+end
+
+##
+#     Initialization of variables
+##
+"""
+    init_state_xyz! 
+    - sets up the initial fields within our state variables
+    (e.g., prognostic, auxiliary, etc.), however it seems to not initialized
+    the gradient flux variables by default.
+"""
+
+function nodal_init_state_auxiliary!(
+    m::ModelSetup,
+    state_auxiliary,
+    tmp,
+    geom,
+)
+    init_state_auxiliary!(m, m.physics.orientation, state_auxiliary, geom)
+
+    state_auxiliary.T_sfc = 0
```

Comment:
is this consistent with the IC of the land model?

Reply:
Yes, at the coupled boundary, `aux.T_sfc` (of Atmos) is set to be equal to `state.T_sfc` (of Land) at the beginning of each Atmos coupling cycle. This line just initializes the elements of the MPIStateArray that are not at the boundary (so they should never be used). Technically, this line should not be needed but I was occasionally getting Nans if the whole field was not initialized. 🤷